changelog/150.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:breaking
```

Comment:
I wonder if we should also include an improvement note for this PR. There was also a lot done to improve the extensibility of the code going forward. I know it won't directly impact users, but it may warrant a blurb in the 0.4.0 changelog.

Reply:
True - the changes ought to make contributing runners quite a bit easier!